packages/sdk/src/cosmwasmclient.ts

```diff
@@ -279,23 +279,23 @@ export class CosmWasmClient {
       (entry): Code => ({
         id: entry.id,
         creator: entry.creator,
-        checksum: Encoding.toHex(Encoding.fromHex(entry.code_hash)),
+        checksum: Encoding.toHex(Encoding.fromHex(entry.data_hash)),
         source: entry.source || undefined,
         builder: entry.builder || undefined,
       }),
     );
   }
 
   public async getCodeDetails(codeId: number): Promise<CodeDetails> {
-    // TODO: implement as one request when https://github.com/cosmwasm/wasmd/issues/90 is done
-    const [codeInfos, getCodeResult] = await Promise.all([this.getCodes(), this.restClient.getCode(codeId)]);
-
-    const codeInfo = codeInfos.find(code => code.id === codeId);
-    if (!codeInfo) throw new Error("No code info found");
+    const getCodeResult = await this.restClient.getCode(codeId);
```

Comment:
Nice. I didn't realize you had already implemented the api using two rest calls (smart).

Glad to get this simpler client side